cmd/buildclient/main.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2022, Sylabs, Inc. All rights reserved.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"github.com/sylabs/scs-build-client/internal/app/buildclient"
+)
+
+const (
+	keyAccessToken      = "auth-token"
+	keySkipTLSVerify    = "skip-verify"
+	keyArch             = "arch"
+	keyArtifactFileName = "output"
+	keyFrontendURL      = "url"
+	keyForceOverwrite   = "force"
+	keyImageSpec        = "image-spec"
+)
+
+var Usage = func() {
+	fmt.Fprintf(os.Stderr, `
+Usage: buildclient [opts] <path to build definition>
+
+    Build and push artifact to cloud library:
+
+        buildclient --image-spec library://user/project/image:tag alpine.def
+
+    Build, push artfifact to cloud library and download locally:
+
+        buildclient --image-spec library://user/project/image:tag --output /tmp/alpine_3.sif alpine.def
```

Comment:
I think we might need a top level command (`buildclient build ...` instead of simply `buildclient ...`) of , as we're almost certainly going to need more down the road. Thinking of a common workflow that we see:

```
buildclient build image.def --output image.sif
// Scan, test, verify, etc.
buildclient sign image.sif
buildclient push image.sif
```

This also gives us the flexibility to add stuff like `buildclient version`, etc.